tests/integration/suite/daprd/conversation/grpc/streaming.go

```diff
@@ -0,0 +1,451 @@
+/*
+Copyright 2025 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(streaming))
+}
+
+type streaming struct {
+	daprd *daprd.Daprd
+}
+
+type liveConversationAIProvider struct {
+	componentName string
+	envVar        string
+}
+
+var liveConversationAIProviders = []liveConversationAIProvider{
+	{
+		componentName: "openai",
+		envVar:        "OPENAI_API_KEY",
+	},
+	{
+		componentName: "anthropic",
+		envVar:        "ANTHROPIC_API_KEY",
+	},
+	{
+		componentName: "gemini",
+		envVar:        "GOOGLE_API_KEY",
+	},
+}
+
+func (s *streaming) Setup(t *testing.T) []framework.Option {
+	// Build component configuration - always include echo, conditionally include AI providers
+	componentConfig := `
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: echo
+spec:
+  type: conversation.echo
+  version: v1
+  metadata:
+  - name: key
+    value: testkey`
+
+	// Define AI provider components
+
+	// Add AI provider components if their API keys are available
+	for _, provider := range liveConversationAIProviders {
+		if apiKey := os.Getenv(provider.envVar); apiKey != "" {
```

Comment:
Move to an end to end test.

Integration tests are always self contained and offline. Use the `echo` conversation component type, or use a mock server.